File: static/js/socket/socket.js
import { SERVER_URL } from "../utils/consts/serverUrl.js";

const ROLE_PILOT = 0;

const socket = io(SERVER_URL, { //! single connection for each pilot
  auth: { r: ROLE_PILOT }
}); 

function send(event, data) {
  socket.emit(event, data);
}

function listen(event, callback) {
  socket.on(event, callback);
}

export { send, listen, socket };

File: app/classes/socket/socket.py
class SocketService:
    def __init__(self, socketio): 
        self.socketio = socketio

    def listen(self, event_name, callback):
        self.socketio.on(event_name)(callback)

    def send(self, event, data, room=None): 
        self.socketio.emit(event, data, room=room)